src/parser/ast.rs

```diff
@@ -55,14 +55,56 @@ pub enum VizType {
 /// A single visualization layer (from DRAW clause)
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct Layer {
-    /// Optional layer name (from AS clause)
-    pub name: Option<String>,
     /// Geometric object type
     pub geom: Geom,
     /// Aesthetic mappings (aesthetic → column or literal)
     pub aesthetics: HashMap<String, AestheticValue>,
     /// Geom parameters (not aesthetic mappings)
     pub parameters: HashMap<String, ParameterValue>,
+    /// Optional filter expression for this layer
+    pub filter: Option<FilterExpression>,
+}
+
+/// Filter expression for layer-specific filtering (from FILTER clause)
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum FilterExpression {
+    /// Logical AND of two expressions
+    And(Box<FilterExpression>, Box<FilterExpression>),
+    /// Logical OR of two expressions
+    Or(Box<FilterExpression>, Box<FilterExpression>),
+    /// Simple comparison
+    Comparison {
+        column: String,
+        operator: ComparisonOp,
+        value: FilterValue,
+    },
+}
+
+/// Comparison operators for filter expressions
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ComparisonOp {
+    /// Equal (=)
+    Eq,
+    /// Not equal (!= or <>)
+    Ne,
+    /// Less than (<)
+    Lt,
+    /// Greater than (>)
+    Gt,
+    /// Less than or equal (<=)
+    Le,
+    /// Greater than or equal (>=)
+    Ge,
+}
+
+/// Values in filter comparisons
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum FilterValue {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+    /// Column reference (for column-to-column comparisons)
+    Column(String),
```

Comment:
This is OK for now, but at some point we'll probably want to expand on how we express filtering in the AST to support other types of comparison operators, matching with `LIKE` etc.

We'll probably need to support filter values for other SQL types like `DATETIME`. That would probably be stored as a string too, but we might need to differentiate from the column-to-column case.

Nothing to worry about right now, but we should open a general "improve filtering" issue for it once we merge.